Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py

```diff
@@ -0,0 +1,578 @@
+
+    outputs = {
+        'AzureSentinel.Entity(val.ID === obj.ID)': flattened_result
+    }
+
+    return (
+        readable_output,
+        outputs,
+        result
+    )
+
+
+def list_entities_command(client, args):
+    # todo: only 20 elements are returned, no skipToken, while there are more (can be found by id).
+    # limit = int(args.get('limit'))
+    # offset = int(args.get('offset'))
+    url_suffix = 'entities'
+
+    result = client.http_request('GET', url_suffix)
+
+    flattened_results = [flatten_entity_attributes(entity) for entity in result.get('value', [])]
+
+    outputs = {
+        'AzureSentinel.Entity(val.ID === obj.ID)': flattened_results
+    }
+
+    # todo: if we don't have $top param, I will do a manual pagination (currently waiting for a session w microsoft guy)
+    readable_output = tableToMarkdown(f'Entities details', flattened_results, removeNull=True)
+
+    return (
+        readable_output,
+        outputs,
+        result
+    )
+
+
+def list_incident_relations_command(client, args):
+    inc_id = args.get('incident_id')
+    top = min(50, int(args.get('top')))
+    next_link = args.get('next_link')
+    entity_kinds = args.get('entity_kinds')
+    filter_expression = args.get('filter', '')
+
+    if next_link:
+        result = client.http_request('GET', full_url=next_link)
+    else:
+        # Handle entity kinds to filter by
+        if entity_kinds:
+            if filter_expression:
+                filter_expression += ' and '
+            filter_expression += f"search.in(properties/relatedResourceKind, '{entity_kinds}', ',')"
+
+        url_suffix = f'incidents/{inc_id}/relations'
+        params = {
+            '$top': top,
+            '$filter': filter_expression
+        }
+        remove_nulls_from_dictionary(params)
+
+        result = client.http_request('GET', url_suffix, params=params)
+
+    relations = [related_resource_data_to_demisto_format(resource, inc_id) for resource in result.get('value')]
+
+    outputs = {f'AzureSentinel.IncidentRelatedResource(val.ID === obj.ID && val.IncidentID == {inc_id})': relations}
+
+    next_link = result.get('nextLink')
+    if next_link:
+        outputs['AzureSentinel.NextLink'] = next_link
+
+    readable_output = tableToMarkdown(f'Incident {inc_id} Relations ({len(relations)} results)', relations,
+                                      headerTransform=pascalToSpace,
+                                      removeNull=True)
+
+    return (
+        readable_output,
+        outputs,
+        result
+    )
+
+
+def fetch_incidents(client, last_run, first_fetch_time):
+    # Get the last fetch details, if exist
+    last_fetch_time = last_run.get('last_fetch_time')
+    last_fetch_ids = last_run.get('last_fetch_ids', [])
+
+    # Handle first time fetch
+    if last_fetch_time is None:
+        last_fetch_time_str, _ = parse_date_range(first_fetch_time, DATE_FORMAT)
+        last_fetch_time = datetime.strptime(last_fetch_time_str, DATE_FORMAT)
```

Comment:
better to use `dateutil.parser.parse` or `dateparser.parse` instead of `datetime.strptime`

Reply:
done